lib/job.js

```diff
@@ -131,6 +131,28 @@ class Job extends BaseModel {
 
         return factory.list(listConfig);
     }
+
+    /**
+     * Remove all builds associated with this job and the job itself
+     * @return {Promise}        Resolves to null if remove successfully
+     */
+    remove() {
+        const removeBuilds = (() =>
+            // Iterate through the builds and remove them
+            this.getBuilds().then(builds => {
+                if (builds.length > 0) {
+                    builds.forEach(build => Promise.all(build.remove()));
```

Comment:
You are both right, in a fashion. `Promise.all` takes an array of promises and waits for them all to be resolved.
```
this.getBuilds().then(builds => {
    if (builds.length === 0) {
        return null; // or super.remove();
    }
    return Promise.all(builds.map(build => build.remove())))
        .then(() => removeBuilds());
```

Same sort of code in pipeline for jobs.